parts/k8s/addons/azure-policy-deployment.yaml

```diff
@@ -77,261 +83,243 @@ spec:
   - list
+  - patch
+  - update
   - watch
 - apiGroups:
-  - ""
+  - templates.gatekeeper.sh
   resources:
-  - configmaps
+  - constrainttemplates/status
   verbs:
   - get
-  - list
-  - watch
-  - create
-  - update
   - patch
-  - delete
+  - update
 - apiGroups:
   - admissionregistration.k8s.io
+  resourceNames:
+  - gatekeeper-validating-webhook-configuration
   resources:
-  - mutatingwebhookconfigurations
   - validatingwebhookconfigurations
   verbs:
-  - get
-  - list
-  - watch
   - create
-  - update
-  - patch
   - delete
-- apiGroups:
-  - ""
-  resources:
-  - secrets
-  verbs:
   - get
   - list
-  - watch
-  - create
-  - update
   - patch
-  - delete
-- apiGroups:
-  - ""
-  resources:
-  - services
-  verbs:
-  - get
-  - list
-  - watch
-  - create
   - update
-  - patch
-  - delete
+  - watch
 ---
 apiVersion: rbac.authorization.k8s.io/v1
-kind: ClusterRoleBinding
+kind: RoleBinding
 metadata:
-  creationTimestamp: null
+  labels:
+    gatekeeper.sh/system: "yes"
+    kubernetes.io/cluster-service: "true"
+    addonmanager.kubernetes.io/mode: Reconcile
   name: gatekeeper-manager-rolebinding
+  namespace: kube-system
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: gatekeeper-manager-role
+subjects:
+- kind: ServiceAccount
+  name: gatekeeper-admin
+  namespace: kube-system
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
   labels:
+    gatekeeper.sh/system: "yes"
     kubernetes.io/cluster-service: "true"
     addonmanager.kubernetes.io/mode: Reconcile
+  name: gatekeeper-manager-rolebinding
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: ClusterRole
   name: gatekeeper-manager-role
 subjects:
 - kind: ServiceAccount
-  name: default
+  name: gatekeeper-admin
   namespace: kube-system
 ---
 apiVersion: v1
 kind: Secret
 metadata:
-  name: gatekeeper-webhook-server-secret
-  namespace: kube-system
   labels:
+    gatekeeper.sh/system: "yes"
     kubernetes.io/cluster-service: "true"
-    addonmanager.kubernetes.io/mode: Reconcile
+    addonmanager.kubernetes.io/mode: EnsureExists
+  name: gatekeeper-webhook-server-cert
+  namespace: kube-system
 ---
 apiVersion: v1
 kind: Service
 metadata:
   labels:
-    control-plane: controller-manager
-    controller-tools.k8s.io: "1.0"
+    gatekeeper.sh/system: "yes"
     kubernetes.io/cluster-service: "true"
     addonmanager.kubernetes.io/mode: Reconcile
-  name: gatekeeper-controller-manager-service
+  name: gatekeeper-webhook-service
   namespace: kube-system
 spec:
   ports:
   - port: 443
     targetPort: 8443
   selector:
     control-plane: controller-manager
-    controller-tools.k8s.io: "1.0"
+    gatekeeper.sh/system: "yes"
 ---
 apiVersion: apps/v1
-kind: StatefulSet
+kind: Deployment
 metadata:
   labels:
     control-plane: controller-manager
-    controller-tools.k8s.io: "1.0"
+    gatekeeper.sh/system: "yes"
     kubernetes.io/cluster-service: "true"
     addonmanager.kubernetes.io/mode: Reconcile
```

Comment:
Is there ever a reason why a user would manually maintain the addon outside of aks-engine cluster operations? Like, increase replica count? If so, we should set `addonmanager.kubernetes.io/mode` to `EnsureExists`, so that those user-initiated customizations won't be overridden by kube-addon-manager.

Reply:
in this case, we want to curate the deployment arguments as much as possible to prevent any potential issues